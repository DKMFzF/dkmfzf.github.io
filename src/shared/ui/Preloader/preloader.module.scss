.preloader {
	align-items: center;
	display: flex;
	height: 100%;
	justify-content: center;
	left: 50%;
	position: absolute;
	top: 50%;
	transform: translate(-50%, -50%);
	width: 100%;
}

.preloader-circle {
	animation: spin 0.75s infinite linear;
	border: 1px solid;
	border-color: #000 #3e3e3e #989898 #fff;
	border-radius: 50%;
	display: block;
	height: 74px;
	width: 74px;
}

@keyframes spin {
	100% {
		transform: rotate(360deg);
	}
}
