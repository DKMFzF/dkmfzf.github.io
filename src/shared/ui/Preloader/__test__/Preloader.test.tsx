import { Preloader } from '../Preloader';
import { render } from '@testing-library/react';
import { expect } from '@jest/globals';

describe('[Preloader]', () => {
	it('Renders preloader correctly and matches snapshot', () => {
		const { asFragment } = render(<Preloader />);
		expect(asFragment()).toMatchSnapshot();
	});
});
