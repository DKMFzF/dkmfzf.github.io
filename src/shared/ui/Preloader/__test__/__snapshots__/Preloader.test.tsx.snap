// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Preloader] Renders preloader correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="preloader"
  >
    <div
      class="preloader-circle"
    />
  </div>
</DocumentFragment>
`;
